contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -398,17 +398,50 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
-   void system_contract::denynames( const std::vector<name>& patterns ) {
+   checksum256 system_contract::denyhashcalc( const std::vector<name>& patterns ) {
+      check(!patterns.empty(), "Cannot compute hash on empty vector");
+      for (auto n : patterns) {
+         // check that pattern is valid (pattern should not be empty or more than 12 character long)
+         canon_name_t pattern(n);
+         check(pattern.valid(),
+               n.value ? ("Pattern " + n.to_string() + " is not valid") : "Empty patterns are not allowed");
+      }
+      static_assert(sizeof(name) == sizeof(uint64_t));
+      return eosio::sha256(reinterpret_cast<const char*>(patterns.data()), patterns.size() * sizeof(name));
+   }
+
+   void system_contract::denyhashadd( const checksum256& hash ) {
       require_auth( get_self() );
 
+      deny_hash_table dh_table(get_self(), get_self().value);
+      const auto idx = dh_table.get_index<"byhash"_n>();
+      auto itr = idx.find(hash);
+      check(itr == idx.end(), "Trying to add a deny hash which is already present");
+      dh_table.emplace(get_self(), [&](auto& row) { row.hash = hash; });
```

Comment:
You are not using the auto-incrementing number to set the `row.id`. This will overwrite any previously set hash rather than add to it.

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/191/commits/ff8645d74ece95122bcdb8f82967941875fccaac